tests/utils/mod.rs

```diff
@@ -0,0 +1,264 @@
+                let next_member_key_package = key_store
+                    .get_mut(&(
+                        group_config.members[client_id].name,
+                        group_config.ciphersuite,
+                    ))
+                    .unwrap()
+                    .pop()
+                    .unwrap();
+                // Have the initial member create an Add proposal using the new
+                // KeyPackage.
+                let add_proposal = mls_group.create_add_proposal(
+                    group_aad,
+                    initial_credential_bundle,
+                    next_member_key_package,
+                );
+                proposal_list.push(add_proposal);
+            }
+            // Create the commit based on the previously compiled list of
+            // proposals.
+            let (commit_mls_plaintext, welcome_option, _key_package_bundle_option) = mls_group
+                .create_commit(
+                    group_aad,
+                    &initial_credential_bundle,
+                    proposal_list.clone(),
+                    false,
+                )
+                .unwrap();
+            let welcome = welcome_option.unwrap();
+            // Distribute the Welcome message to the other members.
+            for client_id in 1..group_config.members.len() {
+                let new_group_member = test_clients
+                    .get(group_config.members[client_id].name)
+                    .unwrap()
+                    .borrow_mut();
+                // Figure out which key package bundle we should use. This is
+                // a bit ugly and inefficient.
+                let member_secret = welcome
+                    .get_secrets_ref()
+                    .iter()
+                    .find(|x| {
+                        match new_group_member
+                            .key_package_bundles
+                            .borrow()
+                            .iter()
+                            .find(|y| y.get_key_package().hash() == x.key_package_hash)
+                        {
+                            Some(_) => true,
+                            None => false,
+                        }
+                    })
+                    .unwrap();
+                let kpb_position = new_group_member
+                    .key_package_bundles
+                    .borrow()
+                    .iter()
+                    .position(|y| y.get_key_package().hash() == member_secret.key_package_hash)
+                    .unwrap();
+                let key_package_bundle = new_group_member
+                    .key_package_bundles
+                    .borrow_mut()
+                    .remove(kpb_position);
+                // Create the local group state of the new member based on the
+                // Welcome.
+                let new_group =
+                    MlsGroup::new_from_welcome(welcome.clone(), None, key_package_bundle).unwrap();
```

Comment:
New members need a ratchet tree here, since we don't use the Ratchet Tree Extension.